code/model/SiteTree.php

```diff
@@ -1411,29 +1399,23 @@ public function MetaTags($includeTitle = true) {
 		return $tags;
 	}
 
-
 	/**
-	 * Returns the object that contains the content that a user would
-	 * associate with this page.
+	 * Returns the object that contains the content that a user would associate with this page.
 	 *
-	 * Ordinarily, this is just the page itself, but for example on
-	 * RedirectorPages or VirtualPages ContentSource() will return the page
-	 * that is linked to.
+	 * Ordinarily, this is just the page itself, but for example on RedirectorPages or VirtualPages ContentSource() will
+	 * return the page that is linked to.
 	 *
-	 * @return SiteTree The content source.
+	 * @return SiteTree The content source
```

Comment:
However `$this` will always be the subtype, just like `static`. `self` is always the parent type.

Eg, if you have an instance of `Page` this function will actually return a `Page` object, but with `@return self` the IDE will only autocomplete for `SiteTree`. If you use `@return static` then the IDE will autocomplete for `Page`, and if you use `@return $this` then the IDE can actually infer the current instance state for the object returned.